std/sync/sync.neva

```diff
@@ -0,0 +1,17 @@
+// WaitGroup outputs a signal after its internal counter decreases to <= 0.
+//
+// :add increases or decreases the value of the internal counter.
+// WaitGroup will not signal until it received at least one nonnegative value from :add.
+//
+// Example:
+//  nodes { wg WaitGroup }
+//  :start -> (2 -> wg:add)
+//  task1 -> (-1 -> wg:add)
+//  task2 -> (-1 -> wg:add)
+//  wg -> :stop  // Signals after both tasks are done.
+//
+// Example:
+//  nodes { wg WaitGroup }
+//  wg -> :stop  // Blocks forever.
+#extern(wait_group)
+pub component WaitGroup(add int) (sig any)
```

Comment:
What do you think about this API?

```
WaitGroup(count int, sig any) (sig any)
```

It accepts `count` and then each `->sig` decrements by one

That's less flexible API but my personal experience with Go was that I was always `wg.Done()` instead of `wg.Add(-n)`

Reply:
I'm ok with adding a sig inport. 

Your proposed change also makes it impossible to add more numbers to the counter later on?

Reply:
> impossible to add more numbers to the counter later on

Well if we're talking about single "iteration" then yes. You send count and bunch of singals, they work together as a single itarton.

After that however you're free to send new count, any int you want.

Do you feel like it's a problem? Did you have use-cases for this?